src/optimizer/plugins/ConstructInlineInstanceOptimizer.ts

```diff
@@ -72,22 +74,54 @@ export default {
             id = rule.value as string;
           } else if (rule.path === `${basePath}.resourceType`) {
             resourceType = rule.value as string;
-          } else if (
-            new RegExp(`^${escapeRegExp(basePath)}\\.meta\\.profile(\\[0\\])?$`).test(rule.path)
-          ) {
-            profile = rule.value as string;
           } else {
             const inlineInstanceRule = cloneDeep(rule);
             inlineInstanceRule.path = rule.path.replace(`${basePath}.`, '');
+            // if this rule is on meta.profile, save its index to check later
+            if (/^meta\.profile(\[\d+\])?$/.test(inlineInstanceRule.path)) {
+              profileIndices.push(inlineInstanceRules.length);
+            }
             inlineInstanceRules.push(inlineInstanceRule);
           }
         });
 
         let newInstance = new ExportableInstance(
           id ?? `Inline-Instance-for-${resource.id}-${++generatedIdCount}`
         );
+
+        // if a profile is available and the user wants it, try to fish it up and use it as InstanceOf
+        if (
+          (profileIndices.length === 1 && options.metaProfile !== 'none') ||
```

Comment:
Is this the same as `(profileIndices.length === 1 && options.metaProfile === 'only-one')`? And if so would it be more clear to use the positive?

Reply:
It's not exactly the same, since it's also okay if `options.metaProfile` is undefined. This shouldn't ever happen, but I wanted to implement it like this in order to represent the idea that the `only-one` behavior is the default that should happen.

Reply:
Ok, that makes sense.